### DISCUSSION TITLE
Can diff method ignore the array order?

### DISCUSSION THREAD

**Discussion by zhuda** [nlohmann/json#2376] [Q&A] [answered]
Hi, I am using the [diff()](https://nlohmann.github.io/json/doxygen/classnlohmann_1_1basic__json_a1c1f21327df91a4dd6c5f5a107240385.html#a1c1f21327df91a4dd6c5f5a107240385) method to compare two json files and would like to ignore the oder changes in array, is it possible? 
```
group: {
    users: [
       {id: 234, name: John},
       {id: 345, name: Mike}
    ]
}
```
should be equal to:

```
group: {
    users: [
       {id: 345, name: Mike},
       {id: 234, name: John}
    ]
}
```

Thanks!

**Comment by nlohmann** [comment, marked as answer]
No, this is not possible. Best would be if you would sort the array yourself before calling diff.

**Reply by zhuda** [reply]
Thanks. I am new to JSON. Maybe I know this is a illegal request or this could be a reasonable enhancement?

**Reply by nlohmann** [reply]
I don't see the benefit for this, because the diff function should detect differences in the array element order. I think your usecase is rather specific.

**Comment by Bigpet** [comment]
Just for context ignoring the order of all arrays in the hierachy to find the minimal diff is computationally very expensive as soon as you nest a couple arrays.
If you don't set sensible constraints a couple of KB of json can take you minutes to diff.
One such library that tries to offer this is deepdiff and even if its default doesn't do a fully exhaustive search it aready took almost half an hour to process a real-world ca. 300KB json, not a constructed example.

(I know the whole "python is slow" stuff didn't help this case but the point remains that very pathological cases make it hard to compute in a bounded time)
